### PR TITLE
Adding ElementAt with tests and typing BOI.

### PR DESCRIPTION
**Pull request by CooRay**
Yeah BOI

### CODE REVIEW

File: src/ElementAt.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * @template T
+ * @param {number} index
+ * @returns {T}
+ */
+function ElementAt(index) {
+  if (this === undefined) throw 'Value cannot be undefined.';
+  if (this === null) throw 'Value cannot be null.';
+  if (index === null) throw 'Index cannot be null.';
+  if (index === null) throw 'Index cannot be null.';
```
Comment (jreina) [inline src/ElementAt.js]: This line is duplicated

File: test/ElementAt.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const expect = require('chai').expect;
+require('..')();
+
+describe('Array#prototype#ElementAt', function() {
+  const input = [1, 2, 3];
+
+  // it('Should throw an error when input is null', function() {
```
Comment (jreina) [inline test/ElementAt.spec.js]: These tests look important

**Comment by CooRay** [comment]
Fixed those 2 files.  I am not sure what's up with the package.lock.  I looks like it upgraded versions automatically.   I can revert it back if I need to.

**Comment by CooRay** [comment]
Reverted package.lock.json to original.

**Comment by jreina** [comment]
Thank you for your contribution. Happy hacking 🎃

**Pull request merged**